incubator/hnc/pkg/forest/forest.go

```diff
@@ -205,3 +220,83 @@ func (ns *Namespace) IsAncestor(other *Namespace) bool {
 	}
 	return ns.parent.IsAncestor(other)
 }
+
+// ClearConditions clears local conditions in the namespace.
+func (ns *Namespace) ClearConditions(key string) {
+	delete(ns.conditions, key)
+}
+
+// SetCondition adds a condition into the list of conditions for key string, returning
+// true if it does not exist previously.
```

Comment:
I don't think we ever use this return value? If so, maybe take it out?

Reply:
done